components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMRoleManagerV2.java

```diff
@@ -480,6 +480,248 @@ public RoleV2 patchRole(String roleId, Map<String, List<PatchOperation>> patchOp
         return getRole(roleId, requiredAttributes);
     }
 
+    @Override
+    public RoleV2 createRoleMeta(RoleV2 role)
+            throws CharonException, ConflictException, NotImplementedException, BadRequestException {
+
+        try {
+            // Check if the role already exists.
```

Comment:
Shall were remove duplicate methods and create unified private methods for v2 and v2 apis?